### PR TITLE
Internal sso

### PR DESCRIPTION
**Pull request by amontenegro**

### CODE REVIEW

File: orcid-core/src/main/java/org/orcid/core/manager/impl/InternalSSOManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Override
+    public void deleteToken(String orcid, HttpServletRequest request, HttpServletResponse response) {
+        // Delete the DB row
+        internalSSODao.delete(orcid);
+        // Delete the cookie
+        if (request.getCookies() != null) {
+            for (Cookie cookie : request.getCookies()) {
+                if (cookie.getName().equals(COOKIE_NAME)) {
+                    cookie.setMaxAge(0);
```
Comment (rcpeters) [inline orcid-core/src/main/java/org/orcid/core/manager/impl/InternalSSOManagerImpl.java]: Setting max age isn't reliable. You should also set the value to nothing.

File: orcid-core/src/main/java/org/orcid/core/manager/InternalSSOManager.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     * */
+    void deleteToken(String orcid, HttpServletRequest request, HttpServletResponse response);
+    
+    /**
+     * Returns true if the given token is still valid
+     * 
+     * @param orcid
+     * @param encryptedToken
+     * */
+    boolean verifyToken(String orcid, String encryptedToken);
```
Comment (rcpeters) [inline orcid-core/src/main/java/org/orcid/core/manager/InternalSSOManager.java]: encryptedToken? Isn't the token just a RandomString?

Comment (amontenegro) [inline orcid-core/src/main/java/org/orcid/core/manager/InternalSSOManager.java]: haha yeah, it should be just token! Fixed

File: orcid-core/src/main/java/org/orcid/core/manager/impl/InternalSSOManagerImpl.java (hunk 1 above)
Comment (amontenegro) [inline orcid-core/src/main/java/org/orcid/core/manager/impl/InternalSSOManagerImpl.java]: Fixed

**Comment by rcpeters** [comment]
merge monster

**Comment by amontenegro** [comment]
@rcpeters fixed!

**Pull request merged**